src/components/SidebarItem/SidebarItem.js

```diff
@@ -0,0 +1,15 @@
+import './SidebarItem.css';
+import {useState} from "react";
+
+const SidebarItem = ({label, handleOnClick}) => {
+  const [state, setState] = useState('inactive');
+
+  const toggleState = () => {
+      setState(state === 'inactive' ? 'active' : 'inactive');
+  };
+  return(
+    <li className={state} onClick={() => {handleOnClick(); toggleState();}}>{label}</li>
+  )
+}
+
+export default SidebarItem;
```

Comment:
When a callback starts to grow and does more than one thing, it is recommended to put it on a function declaration

```suggestion
import './SidebarItem.css';
import {useState} from "react";

const SidebarItem = ({label, handleOnClick}) => {
  const [state, setState] = useState('inactive');
  
  const toggleState = () => {
      setState(state === 'inactive' ? 'active' : 'inactive');
  };
  
   function handleItemClick() {
      handleOnClick()
      toggleState()
  }
  return(
    <li className={state} onClick={handleItemClick}>{label}</li>
  )
}

export default SidebarItem;
```